Source/ARTRest.m

```diff
@@ -724,10 +724,28 @@ - (ARTLocalDevice *)device {
     return ret;
 }
 
-// Store address of once_token to access it in debug function.
-static dispatch_once_t *device_once_token;
-
 - (ARTLocalDevice *)device_nosync {
```

Comment:
So, it's no more "nosync", I think at this point we should just get rid of this method and use `device` which is scheduled via `deviceAccessQueue`. Because immidiatly re-shedule it in `device` on another queue has no sense. Maybe leave it for compatibility here (which would be a call to `device`), so not to remove call to it across codebase in this PR.

Reply:
It needs to be `nosync` (meaning, as I mentioned below, that it needs to be called on the instance's internal queue), because it calls a `nosync` method (`clientid_nosync`).

Reply:
Are you sure it cannot be just like this:

```
- (ARTLocalDevice *)device {
    __block ARTLocalDevice *ret;
    dispatch_sync(ARTRestInternal.deviceAccessQueue, ^{
        ret = [self deviceWithClientId_onlyCallOnDeviceAccessQueue:self.auth.clientId_nosync];
    });
    return ret;
}

// this is temporary for compatibility
- (ARTLocalDevice *)device_nosync {
    return [self device];
}
```

Reply:
I'm pretty sure. The `_nosync` naming of `clientId_nosync` means (as I understand it) that it needs to be called from _queue, not from `ARTRestInternal.deviceAccessQueue`.